pkg/set/set_data.go

```diff
@@ -260,3 +269,49 @@ func NetipAddrPortsToSetData(addrports []netip.AddrPort) ([]SetData, []SetData,
 func NetipAddrPortToSetData(addrport netip.AddrPort) (SetData, SetData, error) {
 	return SetData{Address: addrport.Addr()}, SetData{Port: uint16(addrport.Port())}, nil
 }
+
+// Convert start and end address bytes to SetData type
+func AddressBytesToSetData(start []byte, end []byte) (SetData, error) {
```

Comment:
Please make sure we have 100% test coverage for the various branches in this function and `PortBytesToSetData`.